src/plugins/discover/public/application/components/discover_legacy.tsx

```diff
@@ -163,6 +165,33 @@ export function DiscoverLegacy({
       : undefined;
   const contentCentered = resultState === 'uninitialized';
 
+  const removeElFromColumns = (el: string, columns: string[]) => {
+    const index = columns.indexOf(el);
+    columns.splice(index, 1);
+    return columns;
+  };
+
+  const getDefaultColumns = (cols: string[]) => {
+    let columns = [...cols];
+    if (useNewFieldsApi) {
+      const index = columns.indexOf('_source');
+      if (index !== -1) {
+        columns = removeElFromColumns('_source', columns);
+      }
+    }
+    if (columns.length === 0 && !useNewFieldsApi) {
+      columns = ['_source'];
+    }
+    return columns;
+  };
```

Comment:
This could be simplified a bit (and furthermore be merged with `getDisplayColumns`, since its a short function and just used once)
```TypeScript
 const getDefaultColumns = (cols: string[]) => {
    if (useNewFieldsApi) {
      return cols.filter((column) => column !== '_source');
    }
    return cols.length === 0 ? ['_source'] : cols;
  };
```
`removeElFromColumns` would be redundant